sql: fix ZM dimensions for pg_extension

otan opened the pull request:
Resolves #70776

Release note (bug fix): Fix Z and M coordinate columns causing a panic
for geometry_columns and geography_columns.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/70788)
<!-- Reviewable:end -->

rafiss commented on pkg/sql/pg_extension.go:
```diff
@@ -76,6 +72,17 @@ func postgisColumnsTablePopulator(
+						switch zm {
+						case geopb.ZShapeTypeFlag | geopb.MShapeTypeFlag:
+							datumNDims = tree.NewDInt(4)
+						case geopb.ZShapeTypeFlag, geopb.MShapeTypeFlag:
```
the first case has a `|` and the second case has a `,` -- is that intentional?

otan replied:
oh wait no, this is right. if Z&M are both set, i.e. `|`, it's 4, otherwise 3, etc.

rafiss replied:
sneaky

otan committed:
sql: fix ZM dimensions for pg_extension

Release note (bug fix): Fix Z and M coordinate columns causing a panic
for geometry_columns and geography_columns.

rafiss reviewed (approved):
lgtm! maybe the bitwise OR could use a comment but up to you

otan commented:
bors r=rafiss

otan commented:
flake

bors r=rafiss
